relay-server/src/actors/events.rs

```diff
@@ -1483,12 +1483,13 @@ impl Handler<HandleEnvelope> for EventManager {
 
         let scoping = Rc::new(RefCell::new(envelope.meta().get_partial_scoping()));
         let is_received = Rc::new(AtomicBool::from(false));
+        let envelope_summary = Rc::new(RefCell::new(EnvelopeSummary::empty()));
```

Comment:
You should be able to get rid of the `event_category` variable as a result of this, and use `envelope_summary` in the `or_else` error handler.